packages/ti_recordedfuture/data_stream/playbook_alert/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,639 @@
+---
+description: Pipeline for processing Playbook Alerts.
+processors:
+  - set:
+      field: ecs.version
+      tag: set_ecs_version
+      value: 8.17.0
+  - set:
+      field: event.kind
+      tag: set_alert_to_event_kind
+      value: alert
+  - fail:
+      tag: data_collection_error
+      if: ctx.error?.message != null && ctx.message == null && ctx.event?.original == null
+      message: error message set and no data to process.
+  - rename:
+      field: message
+      tag: rename_message_to_event_original
+      target_field: event.original
+      ignore_missing: true
+      description: Renames the original `message` field to `event.original` to store a copy of the original message. The `event.original` field is not touched if the document already has one; it may happen when Logstash sends the document.
+      if: ctx.event?.original == null
+  - remove:
+      field: message
+      tag: remove_message
+      ignore_missing: true
+      description: The `message` field is no longer required if the document has an `event.original` field.
+      if: ctx.event?.original != null
+  - json:
+      field: event.original
+      tag: json_event_original
+      target_field: recordedfuture.playbook_alert
+      if: ctx.event?.original != null
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+  - script:
+      lang: painless
+      tag: set_panel_status_target_str
+      description: Target field may be list of objects or strings hence this will rename if it is a string.
+      if: ctx.recordedfuture?.playbook_alert?.panel_status?.targets instanceof List
+      source: |-
+        def targets = ctx.recordedfuture.playbook_alert.panel_status.targets;
+        if (targets instanceof List && targets.size() > 0 && targets[0] instanceof String) {
+          ctx.recordedfuture.playbook_alert.panel_status.targets_str = targets;
+          ctx.recordedfuture.playbook_alert.panel_status.remove("targets");
+        }
+  - script:
+      lang: painless
+      tag: set_changes_field
+      description: Target field may be list of objects or strings hence this will rename if it is a string.
+      if: ctx.recordedfuture?.playbook_alert?.panel_log_v2 instanceof List
+      source: |-
+        def panel_log= ctx.recordedfuture.playbook_alert.panel_log_v2;
+        for (panel in panel_log) {
+            if (panel.containsKey("changes") && panel.changes instanceof List) {
+                for (change in panel.changes) {
+                    if (change.containsKey("old") && change.old instanceof String) {
+                        change.old_str = change.old;
+                        change.remove("old");
```

```suggestion
                        change.remove("old");
                        continue;
```
?

(similar below)